### PR TITLE
[EBPF] attacher: Add telemetry for procfs retries

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

There is code to retry procfs reads for up to 10ms after receiving exec events "because procfs could be slow to update". But this makes handling of short-lived processes much slower and leads to the callback channels getting full and missing events.

It is hard to see how procfs could ever be inaccessible after an exec event is received (provided the process is still running). Add some telemetry to check if this case ever hits or if it can be removed.

### Motivation

https://datadoghq.atlassian.net/browse/USMON-694
https://datadoghq.atlassian.net/wiki/spaces/UT/pages/4218553435/#Full-channels-with-larger-pending-callbacks-channel-size

### Describe how to test/QA your changes

Deployed on one staging cluster and checked metrics. Only `read_fail` and `read_success` were seen thus far, `read_temporary_fail` has not been seen as of writing.

### CODE REVIEW

File: pkg/ebpf/uprobes/procfs.go [hunk 1]
```diff
@@ -48,14 +54,26 @@ func waitUntilSucceeds[T any](p *ProcInfo, procFile string, readFunc func(string
...
 		if err != nil {
+			failedTimes++
 			time.Sleep(procFSUpdateInterval)
 		}
 	}
 
+	if err == nil {
+		if failedTimes > 0 {
+			readTemporaryFail.Add(int64(failedTimes))
+		}
```
Comment (guyarb) [inline pkg/ebpf/uprobes/procfs.go:70]: IIRC, `Add` verifies the input parameter is greater than 0, so you can simply remove if-clause

File: pkg/ebpf/uprobes/procfs.go [hunk 2]
```diff
@@ -48,14 +54,26 @@ func waitUntilSucceeds[T any](p *ProcInfo, procFile string, readFunc func(string
...
+
+		readSuccess.Add(1)
+	} else {
+		readFail.Add(1)
```
Comment (guyarb) [inline pkg/ebpf/uprobes/procfs.go:74]: why having 2 metrics, over a single metric with `tag:success` and `tag:failed`?

**Comment by usamasaqib** [comment]
I am a bit confused from the PR description about what this PR is supposed to do. Are you just adding telemetry for how often we failed/succeeded to find a process in procfs?
Or are you also removing the repeated reads on procfs?

**Comment by vitkyrka** [comment]
> I am a bit confused from the PR description about what this PR is supposed to do. Are you just adding telemetry for how often we failed/succeeded to find a process in procfs? Or are you also removing the repeated reads on procfs?

This PR only adds telemetry. The telemetry is mainly to check if the `read_temporary_fail` ever hits. 
If it doesn't hit, the retry + sleep loop can be removed.

File: pkg/ebpf/uprobes/procfs.go (hunk 1 above)
Comment (vitkyrka) [reply pkg/ebpf/uprobes/procfs.go]: Done.

File: pkg/ebpf/uprobes/procfs.go (hunk 2 above)
Comment (vitkyrka) [reply pkg/ebpf/uprobes/procfs.go:72]: No reason, changed.

**Review by guyarb — APPROVED** [review]

**Review by usamasaqib — APPROVED** [review]

**Pull request merged**